esp/smc/SMCLib/TpCommon.cpp

```diff
@@ -179,31 +208,40 @@ static SecAccessFlags getDropZoneScopePermissions(IEspContext& context, const IP
     return queryDistributedFileDirectory().getDropZoneScopePermissions(name, dropZonePath, userDesc);
 }
 
-extern TPWRAPPER_API SecAccessFlags getDZPathScopePermissions(IEspContext& context, const char* dropZoneName, const char* dropZonePath, const char* dropZoneHost)
+extern TPWRAPPER_API SecAccessFlags getDZPathScopePermissions(IEspContext& context, const char* dropZoneName, const char* dropZonePath)
 {
     if (isEmptyString(dropZonePath))
         throw makeStringException(ECLWATCH_INVALID_CLUSTER_NAME, "getDZPathScopePermissions(): DropZone path must be specified.");
 
-    Owned<IPropertyTree> dropZone;
-    if (isEmptyString(dropZoneName))
-        dropZone.setown(findDropZonePlane(dropZonePath, dropZoneHost, true, true));
-    else
-    {
-        dropZone.setown(getDropZonePlane(dropZoneName));
-        if (!dropZone)
-            throw makeStringExceptionV(ECLWATCH_INVALID_INPUT, "getDZPathScopePermissions(): DropZone %s not found.", dropZoneName);
-    }
+    Owned<IPropertyTree> dropZone = getDropZonePlane(dropZoneName);
+    if (!dropZone)
+        throw makeStringExceptionV(ECLWATCH_INVALID_INPUT, "getDZPathScopePermissions(): DropZone %s not found.", dropZoneName);
 
     return getDropZoneScopePermissions(context, dropZone, dropZonePath);
 }
 
-extern TPWRAPPER_API SecAccessFlags getDZFileScopePermissions(IEspContext& context, const char* dropZoneName, const char* dropZonePath,
-    const char* dropZoneHost)
+extern TPWRAPPER_API void checkDZPathScopePermissions(IEspContext& context, const char* dropZoneName, const char* dropZonePath, SecAccessFlags accessReq)
+{
+    SecAccessFlags access = getDZPathScopePermissions(context, dropZoneName, dropZonePath);
+    if (access < accessReq)
+        throw makeStringExceptionV(ECLWATCH_INVALID_INPUT, "Access DropZone Scope %s %s not allowed for user %s (permission:%s). %s Access Required.",
+            dropZoneName, dropZonePath, context.queryUserId(), getSecAccessFlagName(access), getSecAccessFlagName(accessReq));
+}
+
+extern TPWRAPPER_API SecAccessFlags getDZFileScopePermissions(IEspContext& context, const char* dropZoneName, const char* dropZonePath)
```

Comment:
can you change 'dropZonePath' to e.g. 'dropZoneFilePath' to make it clear it's not just a path?
so it is differentiated from getDZPathScopePermissions which should be passed a path (not a path+filename).
(and in header too)

Same for checkDZFileScopePermissions.
Also the errors should clarify what is the scope being rejected in this case, as it stands it will include the filename, but the error says 'Access DropZone Scope' on both, which will lead the user to think a scope is needed for the filename.

Reply:
changed